src/ResultRetriever.cpp

```diff
@@ -141,9 +143,12 @@ ResultRetriever::on_entry_start(uint32_t entry_number,
 void
 ResultRetriever::on_entry_data(const uint8_t* data, size_t size)
 {
-  ASSERT(!(m_dest_file_type == FileType::stderr_output && m_dest_fd));
+  ASSERT(!((m_dest_file_type == FileType::stdout_output
+            || m_dest_file_type == FileType::stderr_output)
+           && m_dest_fd));
 
-  if (m_dest_file_type == FileType::stderr_output
+  if ((m_dest_file_type == FileType::stdout_output
+       || m_dest_file_type == FileType::stderr_output)
```

Comment:
Here you could omit the parenthesis.
Logically it is following: 
- Is it stdout_output ? 
- Or is it stderr_output ? 
- Or is it dependency with non-empty dest path ? 

But from my point of view you can leave it like this.